vllm/entrypoints/openai/api_server.py

```diff
@@ -524,6 +524,17 @@ def init_app_state(
     )
 
 
+def create_server_socket(addr: Tuple[str, int]) -> socket.socket:
+    if socket.has_dualstack_ipv6():
+        sock = socket.create_server(addr,
+                                    family=socket.AF_INET6,
+                                    dualstack_ipv6=True)
```

Comment:
How about single-stack IPv6? I think it's technically possible?

Once you rebase this PR, the host argument here could be an IPv6 address, so this will have to account for that as well.

Reply:
I've noticed that on the main branch, we no longer create a socket and pass it to uvicorn. Therefore, this implementation has reverted to the main branch's approach, with only a simple check for AF_INET6 added. The actual support for IPv4/IPv6/dual-stack is handled within asyncio/uvicorn.